src/test/java/se/iths/jakartaeelab1/service/PersonServiceTest.java

```diff
@@ -0,0 +1,118 @@
+package se.iths.jakartaeelab1.service;
+
+import jakarta.ws.rs.NotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import se.iths.jakartaeelab1.dto.PersonDto;
+import se.iths.jakartaeelab1.entity.Person;
+import se.iths.jakartaeelab1.repository.PersonRepository;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class PersonServiceTest {
+
+    @Mock(lenient = true)
+    private PersonRepository personRepository;
+
+    @InjectMocks
+    private PersonService personService;
+
+    private final UUID testId = UUID.randomUUID();
+    private final Person testPerson = new Person();
+    private final PersonDto testPersonDto = new PersonDto("Test Name", 25, "Test Profession");
+
+    @BeforeEach
+    void setUp() {
+        testPerson.setId(testId);
+        testPerson.setName("Test Name");
+        testPerson.setAge(25);
+        testPerson.setProfession("Test Profession");
+
+        lenient().when(personRepository.updatePerson(any(Person.class))).thenAnswer(invocation -> invocation.getArgument(0));
+        lenient().when(personRepository.findPersonById(testId)).thenReturn(testPerson);
+        lenient().when(personRepository.readAllPerson()).thenReturn(List.of(testPerson));
+        lenient().doNothing().when(personRepository).deletePerson(testId);
+    }
```

Comment:
lenient seems to be deprecated. So maybe we shouldn't be using it. It might be better to resolve the Unnecessary stubbings that mockito is detecting and remake the tests instead of suppressing them? What do you think of this?

Reply:
Agreed, let's avoid lenient and just keep the stubs we need for each test. I'll see what I can do.